src/ButtonComponent/getClasses.ts

```diff
@@ -0,0 +1,139 @@
+    if (size === 'medium') {
+        return {
+            fontSize: '.875rem',
+            padding: '.5rem 1rem'
+        };
+    }
+    if (size === 'large') {
+        return {
+            fontSize: '1rem',
+            padding: '1rem 2rem'
+        };
+    }
+    return {
+        fontSize: '1rem',
+        padding: '1rem 2rem',
+        width: '100%',
+        maxWidth: '30em'
+    };
+};
+
+const getLetterSpacing = (spacing: Spacing) => {
+    if (spacing === 'tracked') return '.1em';
+    if (spacing === 'tight') return '-.05em';
+    if (spacing === 'mega') return '.25em';
+    return '';
+};
+
+const getGrowOnHover = (grow: boolean) => {
+    return (
+        grow && {
+            '-moz-osx-font-smoothing': 'grayscale',
+            '-webkit-backface-visibility': 'hidden',
+            backfaceVisibility: 'hidden',
+            transform: 'translateZ(0)',
+            transition: 'transform .25s ease-out',
+            ':hover': {
+                transform: 'scale(1.05)'
+            },
+            ':focus': {
+                transform: 'scale(1.05)'
+            },
+            ':active': {
+                transform: 'scale(.90)'
+            }
+        }
+    );
+};
+
+export const getClasses = (buttonSettings: BaseButtonStylesConfig) => {
+    const styles: Styles = {
+        button: {
+            backgroundColor: buttonSettings.backgroundColor,
+            borderColor: buttonSettings.borderColor,
+            borderRadius: getBorderRadius(buttonSettings.rounded),
+            color: buttonSettings.textColor || 'white',
+            cursor: 'pointer',
+            display: 'inline-block',
+            fontWeight: buttonSettings.fontWeight,
+            letterSpacing: getLetterSpacing(buttonSettings.letterSpacing),
+            textAlign: 'center',
+            textDecoration: 'none',
+            textTransform: buttonSettings.textTransform,
+            transition: 'color .15s ease-in',
+            ':hover': { transition: 'color .15s ease-in' },
+            ':active': { transition: 'color .15s ease-in' },
+            ':visited': { transition: 'color .15s ease-in' },
+            ':link': { transition: 'color .15s ease-in' },
+            ':focus': {
+                transition: 'color .15s ease-in',
+                outline: '1px dotted currentColor'
+            },
+            ...getBorders(buttonSettings.borderThickness),
+            ...getSizes(buttonSettings.size),
+            ...getGrowOnHover(buttonSettings.growOnHover)
+        },
+        buttonHover: {
+            ':hover': {
+                backgroundColor: buttonSettings.hoverBackgroundColor,
+                borderColor: buttonSettings.hoverBorderColor,
+                color: buttonSettings.hoverTextColor || 'white',
+                textDecoration: 'none',
+                transition: 'all .2s ease-in-out'
+            }
+        }
+    };
+    const classes = StyleSheet.create(styles);
+
+    return `${css(classes.button, classes.buttonHover)}`;
```

Comment:
Just a quibble, but it might be clearer/more idiomatic to return the StyleSheet object here, and put the call to `css()` directly in the `BaseButton` component, like this:

`getClasses.ts`:
```
return StyleSheet.create(styles);
```

`ButtonComponent/index.html`:
```
    const BaseButton: React.FC<BaseButtonProps> = (props) => {
        /* eslint react/prop-types: 0 */
        const classes = getClasses(props.globalButtonSettings);
        return props.href ? (
            <LinkButton {...props} className={css(classes.button, classes.buttonHover)} />
        ) : (
            <StandardButton {...props} className={css(classes.button, classes.buttonHover)} />
        );
    };
```